trestle/core/resolver/modify.py

```diff
@@ -176,14 +176,7 @@ def _add_to_list(input_list: List[OBT], add: prof.Add) -> bool:
         add_list = Modify._add_contents_as_list(add)
         # if by_id is not specified then OSCAL docs say to interpret before and after as starting or ending
         if not add.by_id:
-            # if we are just adding to the list then the add contents should be of the same type
-            if add.position in [prof.Position.before, prof.Position.starting]:
-                for offset, item in enumerate(add_list):
-                    input_list.insert(offset, item)
-                return True
-            else:
-                input_list.extend(add_list)
-                return True
+            raise TrestleError(f'Unexpected add by_id {add.by_id} in Modify.')
```

Comment:
Why is this not allowed anymore?

Reply:
This code was poorly commented and I have cleaned up the comments so they are more clear.  The code here was unreachable because it is only called if by_id is not None.

Reply:
I think this code is needed: https://pages.nist.gov/OSCAL/reference/latest/profile/json-reference/#/profile/modify/alters/adds/position

`by-id` is optional and if not present then modified content will be inserted at the beginning or the end of the list. Or I am missing something and this functionality is now somewhere else?

Reply:
Never mind I just reread the comment, if this code is only called when `by_id` is none then okay